app_new.py

```
@@ -120,31 +120,48 @@ def initialize_session_state():
         st.session_state.selected_db_type = "PostgreSQL"
+    if 'sample_queries' not in st.session_state:
         st.session_state.sample_queries = []
-        
-        # Clear all semantic layer data on app start
+    
+    # Clear all semantic layer data on first app start only
+    if st.session_state.app_initialized and not st.session_state.get('semantic_cleared', False):
         try:
             enhanced_semantic_layer.clear_collections()
+            st.session_state.semantic_cleared = True
         except:
```

Ruff (0.14.2)

Do not use bare `except`

(E722)